src/components/VirtualList/VirtualItem.tsx

```diff
@@ -0,0 +1,76 @@
+import * as React from 'react';
+import { TResource, Resource, IstioTypes } from './Config';
+import { PromisesRegistry } from '../../utils/CancelablePromises';
+import * as Renderer from './Renderer';
+
+type VirtualItemProps = {
+  item: TResource;
+  style: any;
+  className: string;
+  index: number;
+  config: Resource;
+};
+type VirtualItemState = {
+  health?: any;
+};
+
+export default class VirtualItem extends React.Component<VirtualItemProps, VirtualItemState> {
+  private promises = new PromisesRegistry();
+
+  constructor(props: VirtualItemProps) {
+    super(props);
+    this.state = { health: undefined };
+  }
+
+  componentDidMount() {
+    if (this.props.item['healthPromise']) {
+      this.onItemChanged(this.props.item);
+    }
+  }
+
+  componentDidUpdate(prevProps: VirtualItemProps) {
+    if (this.props.item['healthPromise'] && this.props.item['healthPromise'] !== prevProps.item['healthPromise']) {
+      this.onItemChanged(this.props.item);
+    }
+  }
+
+  componentWillUnmount() {
+    this.promises.cancelAll();
+  }
+
+  onItemChanged = async (item: TResource): Promise<void> => {
+    this.promises
+      .register('health', item['healthPromise'])
+      .then(h => {
+        this.setState({ health: h });
+      })
+      .catch(err => {
+        if (!err.isCanceled) {
+          this.setState({ health: undefined });
+          throw err;
+        }
+      });
+  };
+
+  renderDetails = (item: TResource, health?: any) => {
+    const icon = this.getIcon();
+    return this.props.config.columns.map(object => Renderer[object.name](item, this.props.config, icon, health));
```

Comment:
Calling `Renderer[object.name]` is not ideal here as we lose type safety and compiler enforcement on available renderers. But I understand that changing this would lead to more refactoring, so if you want we can look at that later.

Reply:
> @aljesusg when several namespaces are selected and we first enter into a view (e.g. services list), it looks like items are sorted by namespace - but the UI shows they should be sorted by name (the name column has the sort icon highlighted)

Oh yes sorry I know where is the problem, I thought that i upload the file

Reply:
> Calling `Renderer[object.name]` is not ideal here as we lose type safety and compiler enforcement on available renderers. But I understand that changing this would lead to more refactoring, so if you want we can look at that later.

About that it was to be cleaner and simple, only need to add the config and the render method.